test/default_algs.jl

```diff
@@ -35,6 +35,13 @@ solve(prob)
     LinearSolve.OperatorAssumptions(false)).alg ===
       LinearSolve.DefaultAlgorithmChoice.QRFactorization
 
+
+A = spzeros(100, 100)
+A[1,1]=1
+# test that solving a singular problem doesn't error
+prob = LinearProblem(A, ones(100))
+solve(prob)
```

Comment:
This only shows one solver. Can you make KLU and UMFPACK explicit?

Reply:
updated. KLU is still broken (as far as I can see it still doesn't support disabling the checks).

Reply:
@rayegun didn't you handle KLU.jl?

Reply:
https://github.com/SciML/LinearSolve.jl/pull/456#issuecomment-1935228441

https://github.com/search?q=repo%3AJuliaSparse%2FKLU.jl%20check&type=code

You said you added `check=false` to klu but I don't see it in the repo. We can merge for now with this test failing but can you follow up with a test passing verison?